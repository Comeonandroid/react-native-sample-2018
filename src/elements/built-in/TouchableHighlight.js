import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import withPreventDebounce from '../../utils/HOC/withPreventDebounce'


const TouchableHighlight = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

TouchableHighlight.propTypes = {
  children: PropTypes.node
}

const Element = styled.TouchableHighlight``

export default withPreventDebounce(TouchableHighlight)