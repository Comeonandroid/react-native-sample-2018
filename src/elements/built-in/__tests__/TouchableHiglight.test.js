import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import TouchableHighlight from '../TouchableHighlight'

describe('<TouchableHighlight />', () => {
  itBehavesLike('aComponentWithChildren', { subject: <TouchableHighlight /> })
})
