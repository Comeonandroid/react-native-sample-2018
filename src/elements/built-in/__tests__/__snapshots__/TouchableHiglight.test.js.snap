// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TouchableHighlight /> renders component when passed in 1`] = `
<Styled(TouchableHighlight)>
  <Component />
</Styled(TouchableHighlight)>
`;

exports[`<TouchableHighlight /> renders text when passed in 1`] = `
<Styled(TouchableHighlight)>
  Test
</Styled(TouchableHighlight)>
`;
